vsphere/resource_vsphere_license_test.go

```diff
@@ -0,0 +1,253 @@
+
+					%s		 	
+		}`, key, labelString)
+}
+
+func labelToString(labels map[string]string) string {
+	val := ""
+	for key, value := range labels {
+		val += fmt.Sprintf(`
+		label {
+			key = "%s"
+			value = "%s"
+		}
+		`, key, value)
+
+	}
+	return val
+}
+
+func testAccVSphereLicenseBasicCreate() string {
+
+	// precheck already checks if this is present or not
+	key := os.Getenv("VSPHERE_LICENSE")
+
+	return fmt.Sprintf(`resource "vsphere_license" "foo" {
+  		license_key = "%s"
+		}
+	`, key)
+
+}
+
+func testAccVSphereLicenseDestroy(s *terraform.State) error {
+
+	client := testAccProvider.Meta().(*govmomi.Client)
+
+	manager := license.NewManager(client.Client)
+
+	message := ""
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "vsphere_license" {
+			continue
+		}
+
+		key := rs.Primary.ID
+		if isKeyPresent(key, manager) {
+			message += fmt.Sprintf("%s is still present on the server", key)
+		}
+
+	}
+	if message != "" {
+		return errors.New(message)
+	}
+
+	return nil
+}
+
+func testAccVSphereLicenseExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+
+		if !ok {
+			return fmt.Errorf("%s key not found on the server", name)
+		}
+
+		client := testAccProvider.Meta().(*govmomi.Client)
+		manager := license.NewManager(client.Client)
+
+		if !isKeyPresent(rs.Primary.ID, manager) {
+			return fmt.Errorf("%s key not found on the server", rs.Primary.ID)
+		}
+
+		return nil
+	}
+}
+
+func testAccVSphereLicenseNotExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		_, ok := s.RootModule().Resources[name]
+
+		if ok {
+			return fmt.Errorf("%s key should not be present on the server", name)
+		}
+
+		return nil
+	}
+}
+
+func testAccVSpherePreLicenseBasicCheck(t *testing.T) {
+	if key := os.Getenv("VSPHERE_LICENSE"); key == "" {
+		t.Fatal("VSPHERE_LICENSE must be set for acceptance test")
+	}
+}
+
+func testAccVSphereLicenseWithLabelExists(name string) resource.TestCheckFunc {
+
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+
+		if !ok {
+			return fmt.Errorf("%s key not found on the server", name)
+		}
+
+		client := testAccProvider.Meta().(*govmomi.Client)
+		manager := license.NewManager(client.Client)
+
+		if !isKeyPresent(rs.Primary.ID, manager) {
+			return fmt.Errorf("%s key not found on the server", rs.Primary.ID)
+		}
+
+		info := getLicenseInfoFromKey(rs.Primary.ID, manager)
+
+		if len(info.Labels) == 0 {
+			return fmt.Errorf("The labels were not set for the key %s", info.LicenseKey)
+		}
+
+		if len(info.Labels) != 2 {
+			return fmt.Errorf("The number of labels on the server are incorrect", info.Labels)
+		}
+
+		return nil
+	}
+
+}
+
+func TestVSphereLicenseLabelsToMap(t *testing.T) {
+
+	labelMap, err := labelsToMap(labelStub)
+
+	if err != nil {
+		t.Fatal("Error ", err)
+	}
+
+	if value, ok := labelMap["Hello"]; !ok || value != "World" {
```

Comment:
I think it would be better to add a check for the keys that are not being tested here.